### PR TITLE
fix(storage): compactor side reduce sst  meta access  related to non-existent table_id

### PR DESCRIPTION
**Pull request by Li0k**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

as title

1. filter sst related to non-existent table_id before generate_splits
2. avoid some useless meta access in build_delete_range_iter

**This section will be used as the commit message. Please do not leave this empty!**

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- Summarize your change (**mandatory**)
- How does this PR work? Need a brief introduction for the changed logic (optional)
- Describe clearly one logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)
- Refer to a related PR or issue link (optional)

-->

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/storage/src/hummock/compactor/compactor_runner.rs [hunk 1]
```diff
@@ -226,29 +223,57 @@ mod tests {
...
+
+        let sstable_infos = compact_task
+            .input_ssts
+            .iter()
+            .flat_map(|level| level.table_infos.iter())
+            .filter(|table_info| {
+                let table_ids = &table_info.table_ids;
+                table_ids
+                    .iter()
+                    .any(|table_id| compact_task.existing_table_ids.contains(table_id))
```
Comment (soundOfDestiny) [inline src/storage/src/hummock/compactor/compactor_runner.rs:269]: What about sort `existing_table_ids` and use binary search here?

**Review by soundOfDestiny — APPROVED** [review]
LGTM

File: src/storage/src/hummock/compactor/mod.rs [hunk 2]
```diff
@@ -223,11 +225,38 @@ impl Compactor {
...
+            .flat_map(|level| level.table_infos.iter())
+            .filter(|table_info| {
+                let table_ids = &table_info.table_ids;
+                table_ids
+                    .iter()
+                    .any(|table_id| existing_table_ids.contains(table_id))
+            })
+            .cloned()
+            .collect_vec();
+        let compaction_size = compact_task
```
Comment (Little-Wallace) [inline src/storage/src/hummock/compactor/mod.rs:241]: we only need to calculate the size of sstables which not be filtered

File: src/storage/src/hummock/compactor/compactor_runner.rs (hunk 1 above)
Comment (Li0k) [reply src/storage/src/hummock/compactor/compactor_runner.rs:269]: fix it

File: src/storage/src/hummock/compactor/mod.rs (hunk 2 above)
Comment (Li0k) [reply src/storage/src/hummock/compactor/mod.rs]: fix it

**Pull request merged**